src/webgpu/shader/execution/expression/call/builtin/subgroupBallot.spec.ts

```diff
@@ -0,0 +1,339 @@
+    },
+  });
+  const bg = t.device.createBindGroup({
+    layout: pipeline.getBindGroupLayout(0),
+    entries: [
+      {
+        binding: 0,
+        resource: {
+          buffer: sizeBuffer,
+        },
+      },
+      {
+        binding: 1,
+        resource: {
+          buffer: outputBuffer,
+        },
+      },
+    ],
+  });
+
+  const encoder = t.device.createCommandEncoder();
+  const pass = encoder.beginComputePass();
+  pass.setPipeline(pipeline);
+  pass.setBindGroup(0, bg);
+  pass.dispatchWorkgroups(1, 1, 1);
+  pass.end();
+  t.queue.submit([encoder.finish()]);
+
+  const sizeReadback = await t.readGPUBufferRangeTyped(sizeBuffer, {
+    srcByteOffset: 0,
+    type: Uint32Array,
+    typedLength: 1,
+    method: 'copy',
+  });
+  const subgroupSize = sizeReadback.data[0];
+
+  const outputReadback = await t.readGPUBufferRangeTyped(outputBuffer, {
+    srcByteOffset: 0,
+    type: Uint32Array,
+    typedLength: outputNumInts,
+    method: 'copy',
+  });
+  const output = outputReadback.data;
+
+  t.expectOK(checkBallots(output, subgroupSize, filter, expect, allActive));
+}
+
+const kCases = {
+  every_even: {
+    cond: `id % 2 == 0`,
+    filter: (id: number, size: number) => {
+      return id % 2 === 0;
+    },
+    expect: (subgroupSize: number) => {
+      const base = BigInt('0x55555555555555555555555555555555');
+      const mask = getMask(subgroupSize);
+      return base & mask;
+    },
+  },
+  every_odd: {
+    cond: `id % 2 == 1`,
+    filter: (id: number, size: number) => {
+      return id % 2 === 1;
+    },
+    expect: (subgroupSize: number) => {
+      const base = BigInt('0xAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA');
+      const mask = getMask(subgroupSize);
+      return base & mask;
+    },
+  },
+  lower_half: {
+    cond: `id < subgroupSize / 2`,
+    filter: (id: number, size: number) => {
+      return id < Math.floor(size / 2);
+    },
+    expect: (size: number) => {
+      return getMask(Math.floor(size / 2));
+    },
+  },
+  upper_half: {
+    cond: `id >= subgroupSize / 2`,
+    filter: (id: number, size: number) => {
+      return id >= Math.floor(size / 2);
+    },
+    expect: (size: number) => {
+      return getMask(Math.floor(size / 2)) << BigInt(Math.floor(size / 2));
+    },
+  },
+  first_two: {
+    cond: `id == 0 || id == 1`,
+    filter: (id: number) => {
+      return id === 0 || id === 1;
+    },
+    expect: (size: number) => {
+      return getMask(2);
+    },
+  },
+};
+
+g.test('compute,split')
+  .desc('Tests ballot in a split subgroup')
+  .params(u => u.combine('case', keysOf(kCases)))
+  .beforeAllSubcases(t => {
+    t.selectDeviceOrSkipTestCase('subgroups' as GPUFeatureName);
+  })
+  .fn(async t => {
+    const testcase = kCases[t.params.case];
+    const wgsl = `
+enable subgroups;
+
+@group(0) @binding(0)
+var<storage, read_write> size : u32;
+
+@group(0) @binding(1)
+var<storage, read_write> output : array<vec4u>;
+
+@compute @workgroup_size(${kInvocations})
```

Comment:
We'll need fragment shader variants of these execution tests. An .unimplemented() will be fine for now.